test/test_head_request.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding; utf-8 -*-
+"""
+test_head_request
+~~~~~~~~~~~~~~~~~
+"""
+import h2.connection
+
+
+class TestHeadRequest(object):
+
+    def test_non_zero_content_and_no_body(self, frame_factory):
+
+        example_request_headers = [
+            (u':authority', u'example.com'),
+            (u':path', u'/'),
+            (u':scheme', u'https'),
+            (u':method', u'HEAD'),
+            (u':content_length', u'1'),
+        ]
+        example_response_headers = [
+            (u':status', u'200'),
+            (u'server', u'fake-serv/0.1.0'),
+        ]
+
+        c = h2.connection.H2Connection()
+        c.initiate_connection()
+        c.send_headers(1, example_request_headers)
+
+        f = frame_factory.build_headers_frame(
+            example_response_headers
+        )
+        events = c.receive_data(f.serialize())
```

Comment:
I'd like this test to make sure it sets `END_STREAM` on the headers frame too, to ensure that we are testing the full state machine. That'll increase our event count to two.